### PR TITLE
Update github-actions

### PR DESCRIPTION
**Pull request by mmorel-35**
Updates github-actions with their most recent version and enable caching of the npm dependencies

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -13,10 +13,11 @@ jobs:
         os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2.3.4
     - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
+      uses: actions/setup-node@v2.3.0
```
Comment (abetomo) [inline .github/workflows/tests.yml:18]: I think `v2` is better than `v2.x.x`.

**Review by abetomo — APPROVED** [review]
Thanks!

File: .github/workflows/tests.yml (hunk 1 above)
Comment (mmorel-35) [reply .github/workflows/tests.yml]: It's done, the workflow is working fine on my repository.

**Review by shadowspawn — APPROVED** [review]
LGTM

**Pull request merged**